tools/stats/upload_external_contrib_stats.py

```diff
@@ -126,17 +127,35 @@ def get_external_pr_data(
     )
     args = parser.parse_args()
     for i in range(args.length):
+        tries = 0
         startdate = args.startDate + datetime.timedelta(days=i)
         data = get_external_pr_data(
             startdate,
             startdate + datetime.timedelta(days=args.period_length),
             period_length=args.period_length,
         )
-        upload_to_s3(
-            bucket_name="torchci-contribution-data",
-            key=f"external_contribution_counts/{str(startdate)}",
-            docs=data,
-        )
+        while tries < MAXIMUM_RETRIES:
+            success = True
+            upload_to_s3(
+                bucket_name="torchci-contribution-data",
+                key=f"external_contribution_counts/{str(startdate)}",
+                docs=data,
+            )
+            uploaded_data = read_from_s3(
```

Comment:
I think you could just check if the key exists on the bucket.  If it's indeed flaky, I don't think any file would be created.  It would be easier than downloading, decoding, then json loading the whole file

Reply:
A file is created, however, it is incomplete. We would get something like this, but the user field would be completely  empty. (Sorry I reuploaded the data as a quick fix, so I can't reproduce it)
<img width="1101" alt="Screenshot 2023-05-15 at 2 57 10 PM" src="https://github.com/pytorch/pytorch/assets/13758638/f8aad7a5-890d-492f-9c79-5a14126f9551">

Reply:
Oh, I see. So it's not about network flaky issue per my original assumption.  It makes sense now